src/services/open-api-client.js

```diff
@@ -44,13 +44,10 @@ class OpenApiClient {
       }
 
       if (opts.region) {
-        const parts = opts.host.split('.');
-
-        // From 'https://api.twilio.com/' to 'https://api.{region}.twilio.com/'
-        if (parts.length > 1 && parts[1] !== opts.region) {
-          parts.splice(1, 0, opts.region);
-          opts.host = parts.join('.');
-        }
+        const domain = opts.host.split('.').slice(-2).join('.');
+        const prefix = opts.host.split('.' + domain)[0];
+        let product = prefix.split('.')[0];
+        opts.host = [product, opts.edge, opts.region, domain].filter(part => part).join('.');
```

Comment:
How is the original edge being preserved if `opts.edge` is empty?

Reply:
That logic is [here](https://github.com/twilio/twilio-cli-core/pull/86/files#diff-acfdff5a80e1bd752425fa37d497eb1aR190). `opts.edge = this.edge || opts.edge` will have it take on the client's set value first, followed by whatever's passed into opts. If `opts.edge` is still undefined by this point, it's because it's supposed to be.